packages/eas-cli/src/commands/update/configure.ts

```diff
@@ -39,297 +25,27 @@ export default class UpdateConfigure extends EasCommand {
   };
 
   async runAsync(): Promise<void> {
-    Log.log(
-      '💡 The following process will configure your project to run EAS Update. These changes only apply to your local project files and you can safely revert them at any time.'
-    );
     const { flags } = await this.parse(UpdateConfigure);
+    const platform = flags.platform as RequestedPlatform;
     const {
       projectConfig: { projectId, exp, projectDir },
       loggedIn: { graphqlClient },
     } = await this.getContextAsync(UpdateConfigure, {
-      nonInteractive: true,
+      nonInteractive: flags['non-interactive'],
     });
-    const platform = flags.platform as RequestedPlatform;
-
-    if (!isExpoUpdatesInstalledOrAvailable(projectDir, exp.sdkVersion)) {
-      await installExpoUpdatesAsync(projectDir);
-    }
 
-    const workflows = await resolveWorkflowPerPlatformAsync(projectDir);
+    Log.log(
+      '💡 The following process will configure your project to run EAS Update. These changes only apply to your local project files and you can safely revert them at any time.'
+    );
 
-    const updatedExp = await configureAppJSONForEASUpdateAsync({
-      projectDir,
-      projectId,
+    await ensureEASUpdatesIsConfiguredAsync(graphqlClient, {
       exp,
-      platform,
-      workflows,
-    });
-
-    await configureNativeFilesForEASUpdateAsync({
-      projectDir,
       projectId,
-      exp: updatedExp,
+      projectDir,
       platform,
-      workflows,
-      graphqlClient,
     });
 
-    Log.addNewLineIfNone();
-    Log.warn(
-      `All builds of your app going forward will be eligible to receive updates published with EAS Update.`
-    );
-
     Log.addNewLineIfNone();
     Log.log(`🎉 Your app is configured to run EAS Update!`);
   }
 }
-
-type ConfigureForEASUpdateArgs = {
-  projectDir: string;
-  exp: ExpoConfig;
-  platform: RequestedPlatform;
-  workflows: Record<Platform, Workflow>;
-  projectId: string;
-};
-
-export async function configureAppJSONForEASUpdateAsync({
-  projectDir,
-  exp,
-  platform,
-  workflows,
-  projectId,
-}: ConfigureForEASUpdateArgs): Promise<ExpoConfig> {
-  // this command is non-interactive in the way it was designed
-  const easUpdateURL = getEASUpdateURL(projectId);
-  const updates = { ...exp.updates, url: easUpdateURL };
-
-  const androidDefaultRuntimeVersion =
-    workflows.android === Workflow.GENERIC
-      ? DEFAULT_BARE_RUNTIME_VERSION
-      : DEFAULT_MANAGED_RUNTIME_VERSION;
-  const iosDefaultRuntimeVersion =
-    workflows.ios === Workflow.GENERIC
-      ? DEFAULT_BARE_RUNTIME_VERSION
-      : DEFAULT_MANAGED_RUNTIME_VERSION;
-
-  const newAndroidRuntimeVersion =
-    exp.android?.runtimeVersion ?? exp.runtimeVersion ?? androidDefaultRuntimeVersion;
-  const newIosRuntimeVersion =
-    exp.ios?.runtimeVersion ?? exp.runtimeVersion ?? iosDefaultRuntimeVersion;
-
-  let newConfig: Partial<ExpoConfig>;
```

Comment:
would be good to add a unit test for just this piece of code that works before and after your refactor

Reply:
Yeah, I like that too. Unfortunately, this method is one of those problematic methods that are really _really_ hard to test without setting up a project from scratch.

It does a lot of things:
  - Detecting if updates is set up in `app.json`
  - Detecting the current type of project (generic vs managed)
  - Make changes relative to the "requested platform" and current `runtimeVersion`
  - Make sure native files are synced relative to "requested platform" and the current type of project
  - Try to log the steps made, relative to the type of project

If we had to write tests for this, there would be so many cases and variables to cover everything. It's also the main reason why I split this out over multiple functions in this PR.

That being said, we can definitely add these tests for the new split functions without having to set up whole projects during testing.